### PR TITLE
[4.0] Fix mail template not found error

### PR DESCRIPTION
**Pull request by wilsonge**
Pull Request for Issue #30956 .

### Summary of Changes
Fixes logic when the table has no error queued up because nothing went wrong - there's just no entry and adds an additional string to reflect that case.

### Testing Instructions
Force a mail template url that doesn't exist JROOT/administrator/index.php?option=com_mails&view=template&layout=edit&template_id=NOSUCH&language=en-GB

### Actual result BEFORE applying this Pull Request
Screenshot 2020-10-07 at 00 20 51

### Expected result AFTER applying this Pull Request
Screenshot 2020-12-18 at 13 29 23

### Documentation Changes Required
None

### CODE REVIEW

**Comment by brianteeman** [comment]
In your screenshot it shows the error twice?

**Comment by wilsonge** [comment]
It does. Strictly this is because we load the item then we load the form (which internally loads the item to bind the data). I’m not sure of a good fix other than something like only loading the form if the item loads successfully. That is also noted in the text above the screenshot ;)

**Comment by brianteeman** [comment]
Sorry - wrote my comment before coffee. Sure someone will report it as a bug eventually

**Review by HLeithner — APPROVED** [review]

**Comment by wilsonge** [comment]
Refactored. I tested again this morning and realised I couldn't create a mail template after this change - I guess that's what comes of testing things at 5am :) This now has a single error message and is more consistent with other components

File: administrator/language/en-GB/com_mails.ini [hunk 1]
```diff
@@ -10,6 +10,7 @@ COM_MAILS_CONFIG_FIELD_ATTACHMENT_FOLDER_LABEL="Attachments Folder"
 COM_MAILS_CONFIG_FIELD_COPY_MAILS_LABEL="Send Copy"
 COM_MAILS_CONFIG_MAIL_OPTIONS="Mails"
 COM_MAILS_CONFIGURATION="Mail Templates: Options"
+COM_MAILS_ERROR_UNHELD_ID="You are not permitted to use that link to directly access that page (Template: %s)."
```
Comment (wilsonge) [inline administrator/language/en-GB/com_mails.ini:13]: New string as `JLIB_APPLICATION_ERROR_UNHELD_ID` requires an integer but com_mails uses strings

**Comment by chmst** [comment]
I have tested this item :white_check_mark: successfully on 2f381eff7de383fdf27f3e232c349a181d0b77f6<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31699">issues.joomla.org/tracker/joomla-cms/31699</a>.</sub>

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 2f381eff7de383fdf27f3e232c349a181d0b77f6<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31699">issues.joomla.org/tracker/joomla-cms/31699</a>.</sub>

**Comment by wilsonge** [comment]
Thanks guys!

**Pull request merged**